jwst/assign_wcs/miri.py

```diff
@@ -134,102 +136,212 @@ def imaging_distortion(input_model, reference_files):
+    "detector", "v2v3" and "world".
+    v2v3 and world each have spatial,spatial,spectral components
+
     Uses the "specwcs" and "distortion" reference files.
 
     """
-    # Setup the frames.
+    # Define the various coordinate frames.
+    # Original detector frame
     detector = cf.Frame2D(name='detector', axes_order=(0, 1), unit=(u.pix, u.pix))
-    spec = cf.SpectralFrame(name='wavelength', axes_order=(2,), unit=(u.micron,),
-                            axes_names=('lambda',))
-    sky = cf.CelestialFrame(reference_frame=coord.ICRS(), name='sky')
-    v2v3_spatial = cf.Frame2D(name='v2v3_spatial', axes_order=(0, 1), unit=(u.arcsec, u.arcsec))
-    v2v3 = cf.CompositeFrame(name="v2v3", frames=[v2v3_spatial, spec])
-    world = cf.CompositeFrame(name="world", frames=[sky, spec])
-
-    # Determine the distortion model.
+    # Spectral component
+    spec = cf.SpectralFrame(name='spec', axes_order=(2,), unit=(u.micron,), axes_names=('lambda',))
+    # v2v3 spatial component
+    v2v3_spatial = cf.Frame2D(name='v2v3_spatial', axes_order=(0, 1), unit=(u.arcsec, u.arcsec), axes_names=('v2', 'v3'))
+    # v2v3 spatial+spectra
+    v2v3 = cf.CompositeFrame([v2v3_spatial, spec], name='v2v3')
+    # 'icrs' frame which is the spatial sky component
+    icrs = cf.CelestialFrame(name='icrs', reference_frame=coord.ICRS(),
+                             axes_order=(0, 1), unit=(u.deg, u.deg), axes_names=('RA', 'DEC'))
+    # Final 'world' composite frame with spatial and spectral components
+    world = cf.CompositeFrame(name="world", frames=[icrs, spec])
+
+    # Create the transforms
+    dettotel = lrs_distortion(input_model, reference_files)
+    v2v3tosky = pointing.v23tosky(input_model)
+    teltosky = v2v3tosky & models.Identity(1)
+
+    # Put the transforms together into a single pipeline
+    pipeline = [(detector, dettotel),
+                (v2v3, teltosky),
+                (world, None)]
+
+    return pipeline
+
+def lrs_distortion(input_model, reference_files):
+    """
+    The LRS-FIXEDSLIT and LRS-SLITLESS WCS pipeline.
+
+    Transform from subarray x,y to v2,v3, lambda using
+    the "specwcs" and "distortion" reference files.
+
+    """
+
+    # subarray to full array transform
     subarray2full = subarray_transform(input_model)
+
+    # full array to v2v3 transform for the ordinary imager
     with DistortionModel(reference_files['distortion']) as dist:
         distortion = dist.model
 
-    # Incorporate the small rotation
-    angle = np.arctan(0.00421924)
-    rotation = models.Rotation2D(angle)
-    distortion = distortion | rotation
-
-    # Load and process the reference data.
+    # Combine models to create subarray to v2v3 distortion
+    if subarray2full is not None:
+        subarray_dist = subarray2full | distortion
+        subarray_dist.inverse = distortion.inverse | subarray2full.inverse
+    else:
+        subarray_dist = distortion
+        subarray_dist.inverst = distortion.inverse
```

Comment:
```suggestion
        subarray_dist.inverse = distortion.inverse
```

Reply:
Agreed this is a typo- please go ahead and fix in whatever manner is easiest.